src/frontend/src/lib/services/addresses.services.ts

```diff
@@ -32,13 +38,30 @@ export const loadAddresses = async (networkIds: NetworkId[]): Promise<ResultSucc
 };
 
 export const loadIdbAddresses = async (): Promise<ResultSuccessReduced<LoadIdbAddressError>> => {
-	const results = await Promise.all([
-		loadIdbBtcAddressMainnet(),
-		loadIdbEthAddress(),
-		loadIdbSolAddressMainnet()
-	]);
+	const promisesList: Promise<ResultSuccess<LoadIdbAddressError>>[] = [];
+
+	if (get(networkBitcoinMainnetEnabled)) {
+		promisesList.push(loadIdbBtcAddressMainnet());
+	}
+
+	if (get(networkEthereumEnabled)) {
+		promisesList.push(loadIdbEthAddress());
+	}
+
+	if (get(networkSolanaMainnetEnabled)) {
+		promisesList.push(loadIdbSolAddressMainnet());
+	}
+
+	let results = await Promise.all(promisesList);
```

Comment:
maybe this? note that I included EVM networks too

```suggestion
const promisesList: Promise<ResultSuccess<LoadIdbAddressError>>[] = [
	...(get(networkBitcoinMainnetEnabled) ? [loadIdbBtcAddressMainnet()] : []),
	...(get(networkEthereumEnabled) ? [loadIdbEthAddress()] : []),
	...(get(networkSolanaMainnetEnabled) ? [loadIdbSolAddressMainnet()] : [])
];

const results: [ResultSuccess<LoadIdbAddressError>, ResultSuccess<LoadIdbAddressError>, ResultSuccess<LoadIdbAddressError>] = await Promise.all(promisesList);
```